src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -273,13 +273,13 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
   if (opt_cachedir && !opt_unified_core)
     dnf_context_set_keep_cache (dnfctx, TRUE);
   /* For compose, always try to refresh metadata; we're used in build servers
-   * where fetching should be cheap. Otherwise, if --cache-only is set, it's
-   * likely an offline developer laptop case, so never refresh.
+   * where fetching should be cheap. Otherwise, if --cache-only is set, well...
+   * today that just basically sets the default of honoring the repo
+   * metadata_expire. We'd need to enhance the libdnf to have a "really never
+   * expire" option.
```

Comment:
Isn't that just `dnf_repo_set_metadata_expire(repo, G_MAXUINT)` ?

Reply:
No, because libdnf takes the minimum; `G_MAXUINT` is really "use metadata_expire".  See discussion in https://github.com/projectatomic/rpm-ostree/pull/1562